### ISSUE TITLE
Limitation in query depth if relationship type and involved labels were already processed before

### ISSUE THREAD

**Issue by meistermeier** [neo4j/sdn-rx#117]
Internally SDN-RX tries to prevent infinite recursion (and resulting stack overflows) by checking if a combination of start label, type and target label as an descriptor for a relationship was already processed.
This creates an unwanted limitation for the querying right now because the query mechanism as it is cannot create another (and another and another and ...) chain.

We have to revisit the creation mechanism for the return part when relationships are involved.

Reproducer test that can be used in the `RepositoryIT`:
```
@Test
void cannotFetchDeepSameLabelsAndTypeRelationships() {

	long petNode1Id;
	long petNode2Id;
	long petNode3Id;

	try (Session session = driver.session()) {
		Record record = session
			.run("CREATE "
				+ "(p1:Pet{name: 'Pet1'})-[:Has]->(p2:Pet{name: 'Pet2'}), "
				+ "(p2)-[:Has]->(p3:Pet{name: 'Pet3'}) "
				+ "RETURN p1, p2, p3").single();

		petNode1Id = record.get("p1").asNode().id();
		petNode2Id = record.get("p2").asNode().id();
		petNode3Id = record.get("p3").asNode().id();
	}

	Pet loadedPet = petRepository.findById(petNode1Id).get();

	Pet comparisonPet2 = new Pet(petNode2Id, "Pet2");
	Pet comparisonPet3 = new Pet(petNode3Id, "Pet3");
	assertThat(loadedPet.getFriends()).containsExactlyInAnyOrder(comparisonPet2);

	Pet pet2 = loadedPet.getFriends().get(loadedPet.getFriends().indexOf(comparisonPet2));
	assertThat(pet2.getFriends()).containsExactly(comparisonPet3);

}
```

**Comment by meistermeier** [comment]
This will get fixed with #111 and corresponding PR #113 . I leave it open for now to make keep the bug visible and avoid duplicates.

**Comment by meistermeier** [comment]
This is closed by #113 
The solution allows following self-references until a certain (right now a fixed value of 5) depth is reached.
